code/modules/research/xenobiology/crossbreeding/reproductive.dm

```diff
@@ -3,47 +3,31 @@ Reproductive extracts:
 	When fed three monkey cubes, produces between
 	1 and 4 normal slime extracts of the same colour.
 */
+
+
 /obj/item/slimecross/reproductive
 	name = "reproductive extract"
 	desc = "It pulses with a strange hunger."
 	icon_state = "reproductive"
 	effect = "reproductive"
 	effect_desc = "When fed monkey cubes it produces more extracts. Bio bag compatible as well."
 	var/extract_type = /obj/item/slime_extract/
-	var/cubes_eaten = 0
+	var/cooldown = 3 SECONDS
 	var/last_produce = 0
-	var/cooldown = 30 // 3 seconds.
+
+/obj/item/slimecross/reproductive/Initialize()
+	. = ..()
+	LoadComponent(/datum/component/storage/concrete/extract_inventory)
 
 /obj/item/slimecross/reproductive/attackby(obj/item/O, mob/user)
 	if((last_produce + cooldown) > world.time)
 		to_chat(user, span_warning("[src] is still digesting!"))
 		return
```

Comment:
```suggestion
		return

	if(length(contents) >= 3) //if for some reason the contents are full, but it didnt digest, attempt to digest again
		to_chat(user, span_warning("[src] appears to be full but is not digesting! Maybe poking it stimulated it to digest."))
		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_CONSUME_CONTENTS)
		return

```

we want feedback if for some ungodly reason the contents are full and fails to digest, also attempts to kickstart its digestion again